File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.io.*;

public class Main {
    public static int n3,n5,want;
    public static void main(String[] arg)throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int n= Integer.parseInt(br.readLine());
        n3=n/3;
        n5=n/5;
        want=-1;

        bw.write(String.valueOf(cal(n)));
        bw.close();


    }
    public static int cal(int n){
        for(int i=n5;i>=0;i--){
            for(int j=0;j<=n3;j++){
                if(5*i+3*j==n){
                    want=i+j;
                    return want;
                }
            }
        }
        return want;
    }
}

File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
n = int(input())
n3=n//3
n5=n//5
want=-1
for i in range(n5,-1,-1):
    for j in range(0,n3+1):
        if 5*i+3*j==n:
            want = i+j
            break
    if(want!=-1):
        break
print(want)